receiver/mysqlreceiver/client.go

```diff
@@ -152,6 +217,229 @@ func (c *mySQLClient) getIndexIoWaitsStats() ([]IndexIoWaitsStats, error) {
 	return stats, nil
 }
 
+func (c *mySQLClient) getReplicaStatusStats() ([]ReplicaStatusStats, error) {
+	query := "SHOW REPLICA STATUS"
+
+	rows, err := c.client.Query(query)
+	if err != nil {
+		me, ok := err.(*mysql.MySQLError)
+		if !ok {
+			return nil, err
+		}
+
+		if me.Number != 1064 {
+			return nil, err
+		}
+
+		// fallback to slave status
+		query = "SHOW SLAVE STATUS"
```

Comment:
This query and its associated metric names were deprecated in MySQL version 8.0.22, which was [released 2 years ago this month](https://dev.mysql.com/doc/refman/8.0/en/show-replica-status.html). The intention to migrate away from the terminology was [announced several months prior to the release](https://dev.mysql.com/blog-archive/mysql-terminology-updates/). 

We currently claim support for version 8.0. I think we should bump that to 8.0.22 in the documentation and not support the deprecated terminology.

Reply:
We have integration dockerfile for `5.7`, and also `5.7` is globally supported until october 2023. Maybe we do not claim support for it, but it is not a big cost (especially when I developed it) and some users could use it. I already used new names for metrics and structures, so there won't be any change in terminology after `5.7` eol

In addition `SHOW` is not a normal `SELECT`, so I think code approach shouldn't change.

I can change comment, to avoid `slave` word, but I would leave support for `5.7`

Reply:
@djaglowski WDYT?